src/blob/index/core.rs

```diff
@@ -322,7 +322,7 @@ where
         match &self.inner {
             State::InMemory(headers) => {
                 debug!("index get any in memory headers: {}", headers.len());
-                Ok(headers.get(key).and_then(|h| h.first()).cloned())
+                Ok(headers.get(key).and_then(|h| h.last()).cloned())
```

Comment:
Please, add a comment on why the `last` is used here. This should help not to forget about it in future